redis-om-spring/src/main/java/com/redis/om/spring/RedisEnhancedKeyValueAdapter.java

```diff
@@ -245,13 +281,37 @@ public <T> T get(Object id, String keyspace, Class<T> type) {
    */
   @Override
   public void deleteAllOf(String keyspace) {
+    Class<?> type = keyspaceToIndexMap.getEntityClassForKeyspace(keyspace);
```

Comment:
The performance implications of this are going to be brutal, I wonder if it wouldn't just make more sense to script a:

```
FT.DROPINDEX indexName DD
FT.CREATE indexName all the args
```

that way you're dropping down to a single round trip and ensuring that you are in fact killing that entire keyspace without any fear or races. Not a great solution but will probably be more performant & robust.